cvapipe_analysis/tools/cluster.py

```diff
@@ -35,6 +36,9 @@ def __init__(self, control):
         self.abs_path_to_cvapipe = Path(
             os.path.abspath(cvapipe_analysis.__file__)).parents[1]
 
+    def extract_step_name(self):
+        return self.rel_path_to_python_file.split("/")[-2]
```

Comment:
A more elegant way to do this using pathlib: `return Path(self.rel_path_to_python_file).parent.name`. Just keep in mind that both this and your approach would return `compute_features` and not the real step name, which is `computefeatures`. I think it is fine for logging stuff.

Reply:
I agree with you. I pushed a slight improvement.